unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/RPC/RPC.cs

```diff
@@ -26,8 +28,32 @@ private async UniTaskVoid LoadRpcModulesAsync(RpcClientPort port)
             emotes = new ClientEmotesKernelService(await port.LoadModule(EmotesKernelServiceCodeGen.ServiceName));
             analytics = new ClientAnalyticsKernelService(await port.LoadModule(AnalyticsKernelServiceCodeGen.ServiceName));
             modulesLoaded.TrySetResult();
+            await this.StartRpc();
         }
 
+        private async UniTask StartRpc()
+        {
+            await UniTask.SwitchToMainThread();
+
+            // this event should be the last one to be executed after initialization
+            // it is used by the kernel to signal "EngineReady" or something like that
+            // to prevent race conditions like "SceneController is not an object",
+            // aka sending events before unity is ready
```

Comment:
We should document all this setup process somewhere, what type of signasl are required between kernel/renderer to start decentraland.

Also bear in mind we are aiming to register `KernelServices` as `IService` in the ServiceLocator, this initialization will be moved (probably to a subsystem in `Main` or similar). In the meantime we can have it here.

Reply:
i think this method lacks of cohesion. Why is it called `StartRpc` but does an analytics system reports saying that should be the last thing to be called? Perhaps a rename: `ReportSystemInfoToAnalytics`